lib/plugins.js

```diff
@@ -541,14 +547,19 @@ function plugins (settings, deviceId) {
     return _.mapKeys(convertKey, mapValuesWithKey(convert, obj))
   }
 
-  function recordTrade (_tradeEntry) {
+  function recordTrade (_tradeEntry, error) {
     const massage = _.flow(
       _.pick(['cryptoCode', 'cryptoAtoms', 'fiatCode', 'type']),
       convertBigNumFields,
       _.mapKeys(_.snakeCase)
     )
 
     const tradeEntry = massage(_tradeEntry)
+
+    if (error) {
+      tradeEntry.error = error.message
+    }
+
```

Comment:
To make this more functional (by not mutating a record), try adding the error in ``massage``. Other than that, looks good!